pkg/ccl/backupccl/restore_span_covering.go

```diff
@@ -760,27 +662,25 @@ func (i *fileSpanStartAndEndKeyIterator) reset() {
 	i.err = nil
 
 	for _, iter := range i.allIters {
-		iter.reset()
-
-		file, ok := iter.next()
-		if err := iter.err(); err != nil {
+		if ok, err := iter.Valid(); err != nil {
 			i.err = err
 			return
+		} else if !ok {
+			continue
```

Comment:
sanity checking that we want to continue here instead of `break`?

Reply:
yep, the loop is to add iterators that are still valid to the heap.